plugins/source/azure/client/client.go

```diff
@@ -4,81 +4,53 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"strings"
 
 	// Import all autorest modules
-	"github.com/Azure/azure-sdk-for-go/profiles/latest/subscription/mgmt/subscription"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
-	"github.com/Azure/go-autorest/autorest"
-	"github.com/Azure/go-autorest/autorest/azure/auth"
-	"github.com/cloudquery/cloudquery/plugins/source/azure/client/services"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/subscription/armsubscription"
 	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugin-sdk/specs"
 	"github.com/rs/zerolog"
 )
 
+var otherNamespaces = []string{}
+
+var allNamespaces = append(autoGeneratedNamespaces, otherNamespaces...)
+
 type Client struct {
-	subscriptions []string
-	logger        zerolog.Logger
+	subscriptions        []string
+	logger               zerolog.Logger
+	registeredNamespaces map[string]map[string]bool
+	resourceGroups       map[string][]*armresources.GenericResourceExpanded
 	// this is set by table client multiplexer
 	SubscriptionId string
-	services       map[string]*services.Services
+	ResourceGroup  string
+	Creds          azcore.TokenCredential
+	Options        *arm.ClientOptions
 }
 
-func getAuth(logger *zerolog.Logger) (autorest.Authorizer, *azidentity.DefaultAzureCredential, error) {
-	// Old-SDK auth with ordering:
-	//  1. Environment
-	//  2. AzureCLI
-	logger.Info().Msg("Trying to authenticate via environment variables")
-	azureAuth, err := auth.NewAuthorizerFromEnvironment()
-	if err != nil {
-		logger.Info().Msg("Trying to authenticate via CLI")
-		azureAuth, err = auth.NewAuthorizerFromCLI()
-		if err != nil {
-			return nil, nil, err
-		}
-	}
-
-	// New-SDK auth; chained credentials with ordering:
-	//  1. Environment
-	//  2. ManagedIdentity
-	//  3. AzureCLI
-	azCred, err := azidentity.NewDefaultAzureCredential(nil)
-	if err != nil {
-		return azureAuth, nil, err
-	}
-
-	return azureAuth, azCred, nil
-}
-
-func getSubscriptions(logger *zerolog.Logger, spec *Spec, azureAuth autorest.Authorizer) ([]string, error) {
+func getSubscriptions(spec *Spec, creds azcore.TokenCredential) ([]string, error) {
 	subscriptions := make([]string, 0)
 	if len(spec.Subscriptions) == 0 {
```

Comment:
This code fails when a user configures subscriptions under `spec`, e.g.:
```yml
kind: source
...
  spec:
    subscriptions: ["id"]
```

I think you need to add back the `else`:

```
else {
		subscriptions = spec.Subscriptions
	}
```